backend/src/apiserver/resource/test_util.go

```diff
@@ -4,7 +4,7 @@ import "github.com/argoproj/argo-workflows/v3/pkg/apis/workflow/v1alpha1"
 
 func AddRuntimeMetadata(wf *v1alpha1.Workflow) {
 	template := wf.Spec.Templates[0]
-	template.Metadata.Annotations = map[string]string{"sidecar.istio.io/inject": "false"}
+	template.Metadata.Annotations = map[string]string{"sidecar.istio.io/inject": "true"}
```

Comment:
adding the false annotation explicitly disables injection.
but we don't need to make this true though, simply removing the annotation makes most istio envs inject the sidecar by default.